client/my-sites/stats/site.jsx

```diff
@@ -267,23 +265,75 @@ class StatsSite extends Component {
 					</div>
 				</div>
 				<JetpackColophon />
+			</>
+		);
+	}
+
+	renderEnableStatsModule() {
+		const { slug, siteId, path } = this.props;
+		return (
+			<EmptyContent
+				illustration="/calypso/images/illustrations/illustration-404.svg"
+				title={ translate( 'Looking for Stats?' ) }
+				line={ translate(
+					'Enable Site Stats below or visit {{a}}Traffic > Site Stats{{/a}} for more options',
+					{
+						components: {
+							a: <a href={ `/marketing/traffic/${ slug }` } />,
+						},
+					}
+				) }
+				action={ translate( 'Enable Site Stats' ) }
+				actionCallback={ () => this.props.enableJetpackStatsModule( siteId, path ) }
+			/>
+		);
+	}
+
+	render() {
+		const { isJetpack, siteId, showEnableStatsModule } = this.props;
+
+		const { period } = this.props.period;
+
+		return (
+			<Main wideLayout={ true }>
+				<QueryKeyringConnections />
+				{ isJetpack && siteId && <QueryJetpackModules siteId={ siteId } /> }
```

Comment:
Is the `siteId` check necessary here? `isJetpack` will never be true if we don't have a `siteId`.

Reply:
nice catch! updated in https://github.com/Automattic/wp-calypso/pull/39004/commits/d362cd010aaa181d01dfa38da96abce0cd8ba47d